### ISSUE TITLE
Can't use Microsoft.CodeAnalysis.Common on .Net 4.5

### ISSUE THREAD

**Issue by vbfox** [fsprojects/Paket#1915]
### Description

The `Microsoft.CodeAnalysis.Common` package reference `System.Reflection.Metadata` conditionally depending on the framework version and paket doesn't seem to handle it well.

### Repro steps

* Use the following dependencies:

```text
redirects: on
source https://nuget.org/api/v2
nuget Microsoft.CodeAnalysis.Common ~> 1.3.2
``` 

* Run `paket install`
* Observe that `paket.lock` has an incorrect framework restriction on the `System.Reflection.Metadata` package making it inaccessible to .Net 4.5 (But the package is compatible and referenced by a package that is also compatible) :

```text
System.Reflection.Metadata (1.3) - framework: >= net46, >= netstandard13
```

It seem to have used only one restriction from `Microsoft.CodeAnalysis.Common` dependencies instead of merging them all :

```text
    Microsoft.CodeAnalysis.Common (1.3.2)
      ...
      System.Reflection.Metadata (>= 1.2) - framework: portable-net45+win8
      System.Reflection.Metadata (>= 1.3) - framework: >= net46, >= netstandard13
```

### Expected behavior

Transitive dependencies are correctly detected and installed in the project.

### Actual behavior

As no reference is created to the transitive dependency the dll isn't included and the code would crash at runtime.

### Known workarounds

Using project.json ?

More seriously, I tried to specify a very specific framework for the dependency :
```text
nuget System.Reflection.Metadata framework: portable-net45+win8
```

But while the lock file now include that one, the msbuild produced still doesn't want to do what I want :

```xml
<When Condition="($(TargetFrameworkProfile) == 'Profile7') Or ($(TargetFrameworkProfile) == 'Profile44')">
```

For now i'll remove the dependency and reference the whole transitive graph by hand.

**Comment by vbfox** [comment]
A small repro if needed : [repro.zip](https://github.com/fsprojects/Paket/files/459677/repro.zip)

**Comment by matthid** [comment]
Uah seems like another one of those portable problems

**Comment by forki** [comment]
Yes work around for now is to reference the transitives via paket as well.
We hope to solve this soon

Am 07.09.2016 20:56 schrieb "Matthias Dittrich" <notifications@github.com>:
